### PR TITLE
RTEMIS-30: Fee details design change.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-30

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Modified Twig.
- Added fee details data.

### CODE REVIEW

File: modules/rte_mis_school/rte_mis_school.module [hunk 1]
```diff
@@ -64,6 +65,47 @@ function rte_mis_school_preprocess_eck_entity(&$variables) {
...
+
+    $education_details = $variables['content']['field_education_details'][0]['#rows'];
+    // Variable to hold $fee details.
+    $fee_value = [];
+
+    foreach ($education_details as $detail) {
+      $education_detail_id = $detail['data-id'];
+      $education_detail = Paragraph::load($education_detail_id);
+      // Fee Detail of each $education detail.
+      $fee_details = $education_detail->get('field_fee_details');
```
[the remark this answers is not in the document]
Comment (RohanSingh-inno) [reply modules/rte_mis_school/rte_mis_school.module]: added.

File: modules/rte_mis_school/rte_mis_school.module [hunk 2]
```diff
@@ -64,6 +65,50 @@ function rte_mis_school_preprocess_eck_entity(&$variables) {
...
     }
+
+    $education_details = $variables['content']['field_education_details'][0]['#rows'];
+    // Variable to hold $fee details.
+    $fee_value = [];
+
+    foreach ($education_details as $detail) {
+      $education_detail_id = $detail['data-id'];
+      $education_detail = Paragraph::load($education_detail_id);
+      if ($education_detail) {
```
Comment (kunal-sin) [inline modules/rte_mis_school/rte_mis_school.module:76]:
```suggestion
      if ($education_detail instanceof ParagraphInterface) {
```

File: modules/rte_mis_school/rte_mis_school.module [hunk 3]
```diff
@@ -64,6 +65,50 @@ function rte_mis_school_preprocess_eck_entity(&$variables) {
...
+
+        // Get the label for education level and medium from the config.
+        $education_level = $rteMisSchoolConfig->get('field_default_options.field_education_level.' . $education_level_id);
+        $medium = $rteMisSchoolConfig->get('field_default_options.field_medium.' . $medium_id);
+
+        // This will store the fee detail for each class.
+        foreach ($fee_details as $value) {
+          // Get the fee detail id.
+          $fee_detail_id = $value->get('target_id')->getValue();
+          $fee_detail = Paragraph::load($fee_detail_id);
```
Comment (kunal-sin) [inline modules/rte_mis_school/rte_mis_school.module:94]: Please add a safe check here.

File: modules/rte_mis_school/rte_mis_school.module [hunk 4]
```diff
@@ -64,6 +65,50 @@ function rte_mis_school_preprocess_eck_entity(&$variables) {
...
+        $fee_details = $education_detail->get('field_fee_details');
+
+        // Load the coonfig.
+        $rteMisSchoolConfig = \Drupal::config('rte_mis_school.settings');
+        $education_level_id = $education_detail->get('field_education_level')[0]->getString();
+        $medium_id = $education_detail->get('field_medium')[0]->getString();
+
+        // Get the label for education level and medium from the config.
+        $education_level = $rteMisSchoolConfig->get('field_default_options.field_education_level.' . $education_level_id);
+        $medium = $rteMisSchoolConfig->get('field_default_options.field_medium.' . $medium_id);
```
Comment (kunal-sin) [inline modules/rte_mis_school/rte_mis_school.module:88]: What if value is not available?

File: modules/rte_mis_school/rte_mis_school.module [hunk 5]
```diff
@@ -64,6 +65,50 @@ function rte_mis_school_preprocess_eck_entity(&$variables) {
...
+
+        // This will store the fee detail for each class.
+        foreach ($fee_details as $value) {
+          // Get the fee detail id.
+          $fee_detail_id = $value->get('target_id')->getValue();
+          $fee_detail = Paragraph::load($fee_detail_id);
+
+          // Class value of each fee detail.
+          $class_id = $fee_detail->get('field_class_list')->getString();
```
Comment (kunal-sin) [inline modules/rte_mis_school/rte_mis_school.module:97]: Safe check?

File: modules/rte_mis_school/rte_mis_school.module (hunk 2 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_school/rte_mis_school.module]: ok

File: modules/rte_mis_school/rte_mis_school.module [hunk 6]
```diff
@@ -64,6 +65,50 @@ function rte_mis_school_preprocess_eck_entity(&$variables) {
...
+    foreach ($education_details as $detail) {
+      $education_detail_id = $detail['data-id'];
+      $education_detail = Paragraph::load($education_detail_id);
+      if ($education_detail) {
+
+        // Fee Detail of each $education detail.
+        $fee_details = $education_detail->get('field_fee_details');
+
+        // Load the coonfig.
+        $rteMisSchoolConfig = \Drupal::config('rte_mis_school.settings');
```
[the remark this answers is not in the document]
Comment (RohanSingh-inno) [reply modules/rte_mis_school/rte_mis_school.module]: ok

File: modules/rte_mis_school/rte_mis_school.module (hunk 3 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_school/rte_mis_school.module]: ok

File: modules/rte_mis_school/rte_mis_school.module (hunk 4 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_school/rte_mis_school.module]: ok

File: modules/rte_mis_school/rte_mis_school.module (hunk 5 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_school/rte_mis_school.module]: sure

**Review by kunal-sin — APPROVED** [review]

**Pull request merged**